packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -36,14 +35,16 @@ export default <RouterConfig> {
       if (to.hash) {
         return { el: to.hash, top: _getHashElementScrollMarginTop(to.hash), behavior }
       }
+      // The route isn't changing so keep current scroll position
```

Comment:
Note that a query might have changed (e.g. `?filter=content` to `?filter=navigation`)

Reply:
Fair, but in that case today's code also wouldn't scroll anywhere because [this won't set `position`](https://github.com/nuxt/nuxt/blob/3d5c845cb2897221d05212f2eaac17b483c928bf/packages/nuxt/src/pages/runtime/router.options.ts#L27-L29) because [`isChangingPage`](https://github.com/nuxt/nuxt/blob/3d5c845cb2897221d05212f2eaac17b483c928bf/packages/nuxt/src/app/components/utils.ts#L20-L45) doesn't look at the query either.

I don't like returning `false` here, because I think that if you click on a link (even if it's the same page you're on) it should _always_ scroll to top, but since this is my first PR into this repo, I didn't want to make functional behavior changes.

My ideal solution is to mimic what happens if you click a regular link, which normally would put you at the top of the page, even if you already are on that page. To do that, I would replace [this chunk of code](https://github.com/nuxt/nuxt/blob/cf27356f9a6de97d0ac01839c670f3db01f6a71d/packages/nuxt/src/pages/runtime/router.options.ts#L30-L40) with:

```
    // Hash routes on the same page, no page hook is fired so resolve here
    if (to.path === from.path) {
      if (to.hash) {
        return { el: to.hash, top: _getHashElementScrollMarginTop(to.hash), behavior }
      }
      // The route isn't changing but we clicked a link, so scroll to top
      return { left: 0, top: 0 }
    }
```

Reply:
Okay, let's revisit that.